README.md

```diff
@@ -431,6 +431,10 @@ Some places to get you started on the codebase:
   - PubSub: https://github.com/libp2p/go-libp2p-pubsub
 - [IPFS : The `Add` command demystified](https://github.com/ipfs/go-ipfs/tree/master/docs/add-code-flow.md)
 
+### Map of IPFS Subsystems
+**WIP**: This is a high-level architecture diagram of the various sub-systems of go-ipfs. To be updated with how they interact. Anyone who has suggestions is welcome to comment [here](https://docs.google.com/drawings/d/1OVpBT2q-NtSJqlPX3buvjYhOnWfdzb85YEsM_njesME/edit) on how we can improve this!
```

Comment:
This is very similar to https://github.com/ipfs/js-ipfs/tree/master/examples#understanding-the-ipfs-stack, any reason to move to Google Docs rather than iterating on the previous frame?

Reply:
The tool used is https://monodraw.helftone.com and the source file is https://github.com/ipfs/js-ipfs/tree/master/img. However, ultimately it is just an ASCII drawing which makes it super cool for READMEs and print.

Tool aside, I think what is most important is to have an accurate description of the subsystems that is shared and recognized across the project, rather than multiple that show an incomplete picture.

For reference, here are the ones that exist:

- The IPFS Architecture https://github.com/ipfs/js-ipfs#ipfs-architecture
- How the CLI, Client Library, HTTP API and IPFS Core interface with each other https://github.com/ipfs/js-ipfs#code-architecture-and-folder-structure
- Bitswap https://github.com/ipfs/js-ipfs-bitswap#structure

Reply:
I disagree - I think having multiple different diagrams focusing on different parts or trying to convey different messages makes a lot of sense. One focuses on folder structure, one focuses on general context for how repos fit together, another deep dives on a specific module, and one is about tracing a call stack. All of these are valuable knowledge sharing tools and they would lose value if merged into one master diagram. If anything, I want to pull out some areas that don't clearly belong in this diagram to narrow it's purpose - so definitely see it coexisting with others, esp on the js side.

Reply:
do LMK if there is a specific area where this drawing is inaccurate in it's current state - happy to remedy that!

Reply:
To clarify, I believe that we are in agreement in which:
- 👍🏽 to having multiple diagram that focus on different specific parts of the stack (i.e. subsystems or interactions)
- 👎🏽 on having multiple diagrams called IPFS Architecture and having them communicate information that is in conflict
- 👎🏽 on having a diagram that focuses on a subsystem that introduces something that is not coherent with the overall picture

Did I get it write?

Reply:
We can just call this the Go-IPFS architecture if you'd like. This diagram _is_ consistent with the go-ipfs architecture.

Reply:
cool - changed the title to specify go-ipfs